internal/server/iigointernal/judiciary_test.go

```diff
@@ -0,0 +1,310 @@
+package iigointernal
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"gonum.org/v1/gonum/mat"
+)
+
+// TestSoftMergeSanctionThreshold checks whether the softMergeSanctionThresholds is able to correctly merge
+// the clients threshols with the default ones
+func TestSoftMergeSanctionThreshold(t *testing.T) {
+	cases := []struct {
+		name              string
+		clientSanctionMap map[roles.IIGOSanctionTier]roles.IIGOSanctionScore
+		expectedVal       map[roles.IIGOSanctionTier]roles.IIGOSanctionScore
+	}{
+		{
+			name: "Basic soft merge",
+			clientSanctionMap: map[roles.IIGOSanctionTier]roles.IIGOSanctionScore{
+				roles.SanctionTier1: roles.IIGOSanctionScore(3),
+			},
+			expectedVal: map[roles.IIGOSanctionTier]roles.IIGOSanctionScore{
+				roles.SanctionTier1: 3,
+				roles.SanctionTier2: 5,
+				roles.SanctionTier3: 10,
+				roles.SanctionTier4: 20,
+				roles.SanctionTier5: 30,
+			},
+		},
+		{
+			name:              "No merge",
+			clientSanctionMap: map[roles.IIGOSanctionTier]roles.IIGOSanctionScore{},
+			expectedVal: map[roles.IIGOSanctionTier]roles.IIGOSanctionScore{
+				roles.SanctionTier1: 1,
+				roles.SanctionTier2: 5,
+				roles.SanctionTier3: 10,
+				roles.SanctionTier4: 20,
+				roles.SanctionTier5: 30,
+			},
+		},
+		{
+			name: "More complicated merge",
+			clientSanctionMap: map[roles.IIGOSanctionTier]roles.IIGOSanctionScore{
+				roles.SanctionTier1: 7,
+				roles.SanctionTier2: 9,
+				roles.SanctionTier5: 400,
+			},
+			expectedVal: map[roles.IIGOSanctionTier]roles.IIGOSanctionScore{
+				roles.SanctionTier1: 7,
+				roles.SanctionTier2: 9,
+				roles.SanctionTier3: 10,
+				roles.SanctionTier4: 20,
+				roles.SanctionTier5: 400,
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			res := softMergeSanctionThresholds(tc.clientSanctionMap)
+			if !reflect.DeepEqual(res, tc.expectedVal) {
+				t.Errorf("Expected final transgressions to be %v got %v", tc.expectedVal, res)
+			}
+		})
+	}
+}
+
+// Utility functions for testing //
+
+func generateRuleStore() map[string]rules.RuleMatrix {
```

Comment:
remove if unused?

Reply:
I believe those are gonna be used in the upcoming PRs, considering the situation I think we can let it go this time